backend/typescript/rest/animalTypeRoutes.ts

```diff
@@ -0,0 +1,103 @@
+import { Router } from "express";
+// import { isAuthorizedByRole } from "../middlewares/auth";
+import { animalTypeRequestDtoValidator } from "../middlewares/validators/animalTypeValidators";
+import AnimalTypeService from "../services/implementations/animalTypeService";
+import {
+  AnimalTypeResponseDTO,
+  IAnimalTypeService,
+} from "../services/interfaces/animalTypeService";
+import { getErrorMessage, NotFoundError } from "../utilities/errorUtils";
+import { sendResponseByMimeType } from "../utilities/responseUtil";
+
+const animalTypeRouter: Router = Router();
+// animalTypeRouter.use(isAuthorizedByRole(new Set(["User", "Admin"])));
+
+const animalTypeService: IAnimalTypeService = new AnimalTypeService();
+
+/* Create AnimalType */
+animalTypeRouter.post("/", animalTypeRequestDtoValidator, async (req, res) => {
+  try {
+    const { body } = req;
+    const newAnimalType = await animalTypeService.createAnimalType({
+      animal_type_name: body.animal_type_name,
+    });
+    res.status(201).json(newAnimalType);
+  } catch (e: unknown) {
+    res.status(500).send("Internal server error occured.");
```

Comment:
Jerry defined this message as a constant in his errorUtils file, so maybe wait until he commits his changes and then use his constant in your error messages